### ISSUE TITLE
Weather logic - LOCATION issue

### ISSUE THREAD

**Issue by BorisKingma** [frksteenhoff/ClimApp#92]
The as is situation checks your current location in a database

- returns closest set of weather reports based on lat x lon coordinates.
- if distance >10km, no weather report is available and a new weather report is requested.

---
suggest to lower the distance criterion to 1km. 

pro = better localisation of weather report
con = increased probability that no historical comparison can be made.

**Comment by frksteenhoff** [comment]
Could we maybe combine it such that the current weather is fetched from your more or less exact location (as we did in the Android version of the app) and then fetch the forecast data as we do it currently?

**Comment by BorisKingma** [comment]
weather forecast and current weather should be about the same coordinate.

the current weather request should also be available from database. The issues about Time and Location are closely linked, but I think they are easier to oversee and decide on as separate issues.

-- for context 

The reason why we stopped using the original android functionality (request current weather ad hoc) is that we foresee that we can go over the fair use policy of Openweathermap. It that happens, all users are affected.

That's why we chose the route of the database and the distance criterion to prevent calls to openweather when not needed. At the same time, we require a time criterion to force a new weather report if the last stored weather forecast may be outdated for the current situation.

We have to keep in mind that requesting _current_ and _forecast_ means 2 api calls to openweathermap per use of the app, leaving us to 30 app uses per minute in total worldwide.
